doc/viper/viper.md

```diff
@@ -0,0 +1,322 @@
+- `--config-path`
+    - Default: `$(pwd)`
+    - EnvVar: `VT_CONFIG_PATH` (parsed exactly like a `$PATH` style shell variable).
+    - FlagType: `StringSlice`
+    - Behavior: Paths for `ReadInConfig` to search.
+- `--config-type`
+    - Default: `""`
+    - EnvVar: `VT_CONFIG_TYPE`
+    - FlagType: `flagutil.StringEnum`
+        - Values: everything contained in `viper.SupportedExts`, case-insensitive.
+    - Behavior: Force viper to use a particular unmarshalling strategy; required if the config file does not have an extension (by default, viper infers the config type from the file extension).
+- `--config-name`
+    - Default: `"vtconfig"`
+    - EnvVar: `VT_CONFIG_NAME`
+    - FlagType: `string`
+    - Behavior: Instructs `ReadInConfig` to only look in `ConfigPaths` for files named with this name (with any supported extension, unless `ConfigType` is also set, in which case only with that extension).
+- `--config-file`
+    - Default: `""`
+    - EnvVar: `VT_CONFIG_FILE`
+    - FlagType: `string`
+    - Behavior: Instructs `ReadInConfig` to search in `ConfigPaths` for explicitly a file with this name. Takes precedence over `ConfigName`.
+- `--config-file-not-found-handling`
+    - Default: `WarnOnConfigFileNotFound`
+    - EnvVar: (none)
+    - FlagType: `string` (options: `IgnoreConfigFileNotFound`, `WarnOnConfigFileNotFound`, `ErrorOnConfigFileNotFound`, `ExitOnConfigFileNotFound`)
+    - Behavior: If viper is unable to locate a config file (based on the other flags here), then `LoadConfig` will:
+        - `Ignore` => do nothing, return no error. Program values will come entirely from defaults, environment variables and flags.
+        - `Warn` => log at the WARNING level, but return no error.
+        - `Error` => log at the ERROR level and return the error back to the caller (usually `servenv`.)
+        - `Exit` => log at the FATAL level, exiting immediately.
+
+For more information on how viper searches for config files, see the [documentation][viper_read_in_config_docs].
+
+If viper was able to locate and load a config file, `LoadConfig` will then configure the dynamic registry to set up a watch on that file, enabling all dynamic values to pick up changes to that file for the remainder of the program's execution.
+If no config file was used, then dynamic values behave exactly like static values (i.e. the dynamic registry copies in the settings loaded into the static registry, but does not set up a file watch).
+
+### Re-persistence for Dynamic Values
+
+Prior to the introduction of viper in Vitess, certain components (such as `vttablet` or `vtgate`) exposed `/debug/env` HTTP endpoints that permitted the user to modify certain configuration parameters at runtime.
+
+This behavior is still supported, and to maintain consistency between update mechanisms, if:
+- A config file was loaded at startup
+- A value is configured with the `Dynamic: true` option
+
+then in-memory updates to that value (via `.Set()`) will be written back to disk.
```

Comment:
Oh because they don't, strictly-speaking, have a "disk" to write to? Maybe we can abstract the `WriteToDisk` to just `WriteTo(w io.Writer)` and then ...... something something inject a writer that does k8s-specific persistence something something

But yeah let's figure that out for version 1.1 of this or, more likely 2.0 (no real versions here, not related to vitess versions, just making things up 😂)

Reply:
k8s / container infrastructure is supposed to be immutable, so config is usually injected using secrets. Here we are building config that is designed to be changed at runtime. I'm not sure how the two go together. But yeah we can create an issue on the operator repo to figure out how (whether) operator supports dynamic config.